Source/FirebaseNotificationsAddon.swift

```diff
@@ -137,7 +160,19 @@ open class FirebaseNotificationsAddon: NSObject, HaloNotificationsAddon, HaloLif
     open func application(_ app: UIApplication, didReceiveRemoteNotification userInfo: [AnyHashable: Any], core: CoreManager, userInteraction user: Bool, fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
 
         let notification = HaloNotification(userInfo: userInfo)
-        
+    
+        let haloNotificationEvent : HaloNotificationEvent = HaloNotificationEvent(device: (core.device?.alias)!, schedule: userInfo["scheduleId"] as! String, action: EventType.receipt.rawValue)
+
+        if(notificationEvents){
+            core.notificationAction(notificationEvent: haloNotificationEvent, completionHandler: { (event, error) in
+                self.checkTwoFactorAndNotify(notification: notification, app: app, user: user, completionHandler: completionHandler)
```

Comment:
To avoid having duplicated code, try to enclose the "checkTwo..." method inside a mini block and call that inside the closure or on the else